### DISCUSSION TITLE
Leaving OverflowMenu open, onclick of action item. Menu should get closed when clicked outside overflowmenu.

### DISCUSSION THREAD

**Discussion by Ankita-Jaiswal** [carbon-design-system/carbon#11077] [Help]
![image](https://user-images.githubusercontent.com/16722905/160079426-13bd0b79-42ec-441f-96be-0926a5973704.png)

We are using DataTable with Toolbar Menu. This menu is basically action items to hide/show columns(achieved through checkbox for each column )and column reordering. So every time, click on the checkbox or arrow up/down to reorder column...it closes the overflow menu. Is there any way to keep this overflow menu open and close it only when the user clicks outside overflow menu

**Comment by tay1orjones** [comment]
@Ankita-Jaiswal Yes, `OverflowMenu` accepts an `open` prop to control the open state and also provides an `onOpen` callback so you can keep track of the open state and control or override `open` as necessary for your use case.

https://github.com/carbon-design-system/carbon/blob/9db9b14761bfaabde5cd54e1f47957d243d90321/packages/react/src/components/OverflowMenu/OverflowMenu.js#L204-L212

**Comment by Ankita-Jaiswal** [comment]
@tay1orjones , its Data table toolbar menu. However, i have tried using open/onOpen/onclick etc, yet on click of action items it's closed and if inspect element, you will observe, that node is removed immediately

**Comment by tay1orjones** [comment]
@Ankita-Jaiswal You'll need to override the `closeMenu` function on `TableToolbarAction`. [Here's an example](https://codesandbox.io/s/dry-night-9ywbqg?file=/src/index.js).
